mono/metadata/object.c

```diff
@@ -7195,21 +7195,19 @@ mono_string_to_utf16 (MonoString *s)
 {
 	MONO_REQ_GC_UNSAFE_MODE;
 
-	char *as;
-
 	if (s == NULL)
 		return NULL;
 
-	as = (char *)g_malloc ((s->length * 2) + 2);
-	as [(s->length * 2)] = '\0';
-	as [(s->length * 2) + 1] = '\0';
+	int const length = s->length;
 
-	if (!s->length) {
-		return (gunichar2 *)(as);
-	}
+	gunichar2 * const as = (gunichar2 *)g_malloc ((length * 2) + 2);
```

Comment:
While you are at it, you should replace `2` by `sizeof(gunichar2)` to avoid "magic" numbers, so you could have `(length + 1) * sizeof(gunichar2)`.

Reply:
I considered it. ok.

Reply:
The 2 is more reasonable than other magic numbers. However perhaps that is me. I'll update.